pkg/users/local.js

```diff
@@ -498,8 +498,22 @@ PageAccountsCreate.prototype = {
 
     create: function() {
         var tasks = [
-            function create_user() {
-                var prog = ["/usr/sbin/useradd", "--create-home", "-s", "/bin/bash"];
+            () => {
+                const dfd = cockpit.defer();
+                cockpit.spawn(["/usr/sbin/useradd", "-D"])
+                        .done(defaults => {
+                            defaults.split("\n").forEach(item => {
+                                if (item.indexOf("SHELL=") === 0) {
+                                    this.user_shell = item.split("=")[1] || "";
+                                }
+                            });
+                            dfd.resolve();
+                        })
+                        .fail(dfd.resolve); // Don't fail if we cannot read defaults
+                return dfd.promise();
+            },
+            () => {
+                var prog = ["/usr/sbin/useradd", "--create-home", "-s", this.user_shell || "/bin/bash"];
```

Comment:
Does the default shell really need to get specified explicitly? I thought if there is an explicit one, useradd would use it, and `-s /bin/bash` would only be necesssary on an empty value? That would simplify the parsing a bit.

Reply:
Sure, we need `-s /bin/bash` when shell is not specified in defaults. I, however, don't see how it would simply the parsing. We still need to parse it the same way and remember if there was some value or not. Then surely we could add just condition for emptiness and append `-s /bin/bash`.

Reply:
OK, fair enough.